store/tikv/backoff.go

```diff
@@ -128,15 +129,24 @@ type Backoffer struct {
 	maxSleep   int
 	totalSleep int
 	errors     []error
+	ctx        context.Context
 }
 
 // NewBackoffer creates a Backoffer with maximum sleep time(in ms).
-func NewBackoffer(maxSleep int) *Backoffer {
+func NewBackoffer(maxSleep int, ctx context.Context) *Backoffer {
 	return &Backoffer{
 		maxSleep: maxSleep,
+		ctx:      ctx,
 	}
 }
 
+// WithCancel returns a cancel function which, when called, would cancel backoffer's context.
+func (b *Backoffer) WithCancel() context.CancelFunc {
+	var cancel context.CancelFunc
+	b.ctx, cancel = context.WithCancel(b.ctx)
```

Comment:
why WithCancel overwrites parent b.ctx? 
We should return the child ctx and cancel function for outer usage.